SparseGrids/tsgGridWavelet.cpp

```diff
@@ -331,6 +366,32 @@ double GridWavelet::evalIntegral(const int p[]) const{
     }
     return v;
 }
+void GridWavelet::evalDiffBasis(const int p[], const double x[], double jacobian[]) const {
+    // Evaluates the derivative of the wavelet basis given at point p at the coordinates given by x.
+    // The derivative function of a symmetric function will be antisymmetric across the origin for every x != 0.0.
+    std::vector<double> value_cache(num_dimensions), derivative_cache(num_dimensions);
+    for(int i=0; i<num_dimensions; i++) {
+        value_cache[i] = rule1D.eval<0>(p[i], x[i]);
+        derivative_cache[i] = rule1D.eval<1>(p[i], x[i]);
```

Comment:
Derivative cache is used only once, why cache it instead of writing to `jacobian` directly. The second loop down will only deal with the contribution of the `value_cache`.

Reply:
Removed the cache in the latest push.